scripts/cached-test-runner.sh

```
@@ -0,0 +1,162 @@
+    
+    # Check if results are cached
+    if "$CACHE_MANAGER" check "$test_file" 2>/dev/null; then
+        echo -e "${CYAN}⚡ Using cached results for: $test_file${NC}"
+        "$CACHE_MANAGER" get "$test_file" > "$output_file"
+        return 0
+    fi
+    
+    # Run the test
+    echo -e "${BLUE}🔄 Running test: $test_file${NC}"
+    
+    local start_time=$(date +%s)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)